src/ScanningSession.cpp

```diff
@@ -71,12 +77,131 @@ struct xccdf_session* ScanningSession::getXCCDFSession() const
     return mSession;
 }
 
-void ScanningSession::openFile(const QString& path)
+void ScanningSession::cleanTmpDir()
+{
+    /*
+     * For use in cloneToTemporaryFile(...); closes mOpenDir
+     * if it is open and removes backing source. Ignores errors.
+     *
+     * A new Temporary Directory will be created afterwards,
+     * so the worst case is leaving temporary directory (+files),
+     * but this is cleaned on reboot or manually by the user.
+     */
+    if (mOpenDir != NULL) {
+        mOpenDir->remove();
+        delete mOpenDir;
+        mOpenDir = NULL;
+    }
+}
+
+void ScanningSession::cloneToTemporaryFile(const QString& path)
```

Comment:
I think that the last `for` loop can be extracted to a private method, as this method is quite complex. It is a pity that we don't have unit tests in the project, because the functionality here would be an ideal candidate.